traffic_ops/traffic_ops_golang/server/servers.go

```diff
@@ -1574,50 +1457,73 @@ func Update(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
-	rows, err := inf.Tx.NamedQuery(updateQuery, server)
-	if err != nil {
-		userErr, sysErr, errCode = api.ParseDBError(err)
+	serverID, errCode, userErr, sysErr := updateServer(inf.Tx, server)
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
+		return
+	}
+
+	if userErr, sysErr, errCode = deleteInterfaces(id, tx); userErr != nil || sysErr != nil {
 		api.HandleErr(w, r, tx, errCode, userErr, sysErr)
 		return
 	}
-	defer rows.Close()
 
-	rowsAffected := 0
-	for rows.Next() {
-		if err := rows.StructScan(&server); err != nil {
-			api.HandleErr(w, r, tx, http.StatusNotFound, nil, fmt.Errorf("scanning lastUpdated from server insert: %v", err))
+	if userErr, sysErr, errCode = createInterfaces(id, server.Interfaces, tx); userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, tx, errCode, userErr, sysErr)
+		return
+	}
+
+	if server.UpdPending != nil && *server.UpdPending { // To continue to work with the legacy implementation and priority. However, consider bool UpdPending deprecated
```

Comment:
This API shouldn't be used to queue/unqueue updates/revals for a server -- that is what `POST /servers/id/update` is for. We shouldn't add this functionality to this API IMO.

Reply:
Hm. I don't disagree, but these fields are part of the server structs. Currently `updPending` is even required, although `revalPending` is not. Whether intentional or not, we provide this functionality currently.

I'm happy to remove this logic. Since we use the same struct for both request and response, I've left the timestamp fields as pointers. This way the clients are not required to set values for those fields when using the library. And we could outright ignore them on PUT/POST/PATCH?